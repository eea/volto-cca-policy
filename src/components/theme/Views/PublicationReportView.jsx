import React from 'react';
import {
  HTMLField,
  ContentMetadata,
  PublishedModifiedInfo,
  DocumentsList,
  ShareInfo,
  ReferenceInfo,
  BannerTitle,
  LogoWrapper,
} from '@eeacms/volto-cca-policy/helpers';
import { Image, Grid, Divider } from 'semantic-ui-react';

function PublicationReportView(props) {
  const { content } = props;
  const { logo, title } = content;

  return (
    <div className="db-item-view publication-report-view">
      <BannerTitle
        content={{ ...content, image: '' }}
        type="Publications and Report"
      />

      <div className="ui container">
        <Grid columns="12">
          <div className="row">
            <Grid.Column
              mobile={12}
              tablet={12}
              computer={8}
              className="col-left"
            >
              <LogoWrapper logo={logo}>
                <h2>Description</h2>
                {logo && (
                  <Image
                    src={logo?.scales?.mini?.download}
                    alt={title}
                    className="db-logo"
                  />
                )}
              </LogoWrapper>
              <HTMLField value={content.long_description} />
              <Divider />
              <ReferenceInfo content={content} />
              <PublishedModifiedInfo {...props} />
              <ShareInfo {...props} />
            </Grid.Column>

            <Grid.Column
              mobile={12}
              tablet={12}
              computer={4}
              className="col-right"
            >
              <ContentMetadata {...props} />

              <DocumentsList {...props} />
            </Grid.Column>
          </div>
        </Grid>
      </div>
    </div>
  );
}

export default PublicationReportView;
