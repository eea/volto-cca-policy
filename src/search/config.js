import { mergeConfig } from '@eeacms/search';

import facets from './facets';

// import views from './views';
// import filters from './filters';
// import vocabs from './vocabulary';

const getClientProxyAddress = () => {
  const url = new URL(window.location);
  url.pathname = '';
  url.search = '';
  return url.toString();
};

const ccaConfig = {
  title: 'ClimateAdapt Main',
};

export default function installMainSearch(config) {
  const envConfig = process.env.RAZZLE_ENV_CONFIG
    ? JSON.parse(process.env.RAZZLE_ENV_CONFIG)
    : ccaConfig;

  const pjson = require('@eeacms/volto-cca-policy/../package.json');

  envConfig.app_name = pjson.name;
  envConfig.app_version = pjson.version;

  config.searchui.ccaSearch = {
    ...mergeConfig(envConfig, config.searchui.globalsearchbase),
    elastic_index: '_es/globalsearch',
    index_name: 'data_searchui',
    host: process.env.RAZZLE_ES_PROXY_ADDR || 'http://localhost:3000',
  };

  const { ccaSearch } = config.searchui;

  ccaSearch.permanentFilters.push({
    term: {
      cluster_name: 'cca',
    },
  });
  ccaSearch.permanentFilters.push({
    term: {
      cca_include_in_mission: 'true',
    },
  });

  ccaSearch.facets = facets;

  ccaSearch.initialView.tilesLandingPageParams.sections = [
    {
      id: 'types',
      title: 'Types',
      facetField: 'objectProvides',
      sortOn: 'alpha',
      icon: {
        family: 'Content types',
      },
    },
    // {
    //   id: 'topics',
    //   title: 'Topics',
    //   facetField: 'topic',
    //   sortOn: 'alpha',
    //   whitelist: [
    //     'Agriculture and food',
    //     'Air pollution',
    //     'Bathing water quality',
    //     'Biodiversity',
    //     'Bioeconomy',
    //     'Buildings and construction',
    //     'Chemicals',
    //     'Circular economy',
    //     'Climate change adaptation',
    //     'Climate change mitigation',
    //     'Electric vehicles',
    //     'Energy',
    //     'Energy efficiency',
    //     'Environmental health impacts',
    //     'Environmental inequalities',
    //     'Extreme weather',
    //     'Fisheries and aquaculture',
    //     'Forests and forestry',
    //     'Industry',
    //     'Land use',
    //     'Nature protection and restoration',
    //     'Nature-based solutions',
    //     'Noise',
    //     'Plastics',
    //     'Pollution',
    //     'Production and consumption',
    //     'Renewable energy',
    //     'Resource use and materials',
    //     'Road transport',
    //     'Seas and coasts',
    //     'Soil',
    //     'Sustainability challenges',
    //     'Sustainability solutions',
    //     'Sustainable finance',
    //     'Textiles',
    //     'Transport and mobility',
    //     'Urban sustainability',
    //     'Waste and recycling',
    //     'Water',
    //   ],
    // },
    // {
    //   id: 'countries',
    //   title: 'Countries',
    //   facetField: 'spatial',
    //   filterType: 'any:exact',
    //   sortOn: 'alpha',
    //   maxPerSection: 100,
    //   whitelist: [
    //     'Austria',
    //     'Belgium',
    //     'Bulgaria',
    //     'Croatia',
    //     'Cyprus',
    //     'Czechia',
    //     'Denmark',
    //     'Estonia',
    //     'Finland',
    //     'France',
    //     'Germany',
    //     'Greece',
    //     'Hungary',
    //     'Iceland',
    //     'Ireland',
    //     'Italy',
    //     'Latvia',
    //     'Liechtenstein',
    //     'Lithuania',
    //     'Luxembourg',
    //     'Malta',
    //     'Netherlands',
    //     'Norway',
    //     'Poland',
    //     'Portugal',
    //     'Romania',
    //     'Slovakia',
    //     'Slovenia',
    //     'Spain',
    //     'Sweden',
    //     'Switzerland',
    //     'TÃ¼rkiye',
    //     'Albania',
    //     'Bosnia and Herzegovina',
    //     'Kosovo',
    //     'Montenegro',
    //     'North Macedonia',
    //     'Serbia',
    //   ],
    //   icon: {
    //     family: 'CountryFlags',
    //     className: 'facet-option-icon',
    //   },
    // },
    // {
    //   id: 'language',
    //   title: 'Languages',
    //   facetField: 'language',
    //   sortOn: 'custom',
    //   sortOrder: 'asc',
    // },
    // {
    //   id: 'website',
    //   title: 'Websites',
    //   facetField: 'cluster_name',
    //   sortOn: 'count',
    //   sortOrder: 'desc',
    //   icon: {
    //     family: 'Sources',
    //     className: 'facet-option-icon',
    //   },
    // },
  ];

  if (typeof window !== 'undefined') {
    config.searchui.ccaSearch.host =
      process.env.RAZZLE_ES_PROXY_ADDR || getClientProxyAddress();
  }
  // console.log(config.searchui.ccaSearch);

  return config;
}
