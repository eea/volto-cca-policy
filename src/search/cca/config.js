import { mergeConfig } from '@eeacms/search';
import { getClientProxyAddress } from '../utils';
import { vocab } from '../vocabulary';

import facets from './facets';
import views from './views';
import { download_fields } from '../../constants';

const ccaConfig = {
  title: 'ClimateAdapt Main',
  ...views,
};

export const clusters = {
  name: 'op_cluster',
  // field: 'objectProvides',
  clusters: [
    {
      name: 'Climate-ADAPT',
      icon: { name: 'file text' },
      values: ['climate'],
    },
    {
      name: 'Health Observatory',
      icon: { name: 'compass' },
      values: ['observatory'],
    },
    {
      name: 'Mission Portal',
      icon: { name: 'compass' },
      values: ['mission'],
    },
  ].map((cluster) => ({
    ...cluster,
    defaultResultView: 'ClusterHorizontalCardItem',
  })),
};

const cca_build_runtime_mappings = {};
cca_build_runtime_mappings['op_cluster'] = {
  type: 'keyword',
  script: {
    source:
      "def objectProvides = ['Adaptation option','Case study','Guidance','Video','Indicator','Information portal','Organisation','Publication reference','Research and knowledge project','Video','Tool'];def vals = doc['objectProvides'];for (defVal in objectProvides) {for (objVal in vals) {if (objVal.contains(defVal)) {emit('Climate-ADAPT')}}}emit('_all_');if (doc['cca_include_in_search_observatory'][0]) {for (defVal in objectProvides) {for (objVal in vals) {if (objVal.contains(defVal)){emit('Health Observatory')}}}}  if (doc['cca_include_in_mission.keyword'][0].toString() == 'true') {emit('Mission Portal')} else {if (doc['id.keyword'].length==1) { if(doc['id.keyword'][0].toString().contains('climate-adapt.eea.europa.eu/en/mission')) {emit('Mission Portal')}}}",
  },
};

export default function installMainSearch(config) {
  const envConfig = ccaConfig;
  const pjson = require('@eeacms/volto-cca-policy/../package.json');

  envConfig.app_name = pjson.name;
  envConfig.app_version = pjson.version;

  config.searchui.ccaSearch = {
    ...mergeConfig(envConfig, config.searchui.globalsearchbase),
    elastic_index: '_es/globalsearch',
    index_name: 'data_searchui',
    host: process.env.RAZZLE_ES_PROXY_ADDR || 'http://localhost:3000',
    runtime_mappings: cca_build_runtime_mappings,
    vocab,
  };

  config.searchui.ccaSearch.download_fields = download_fields;

  const { ccaSearch } = config.searchui;

  ccaSearch.contentSectionsParams = {
    enable: true,
    sectionFacetsField: 'op_cluster',
    sections: clusters.clusters,
    clusterMapping: Object.assign(
      {},
      ...clusters.clusters.map(({ name, values }) =>
        Object.assign({}, ...values.map((v) => ({ [v]: name }))),
      ),
    ),
  };

  ccaSearch.permanentFilters.push({
    term: {
      cluster_name: 'cca',
    },
  });

  ccaSearch.facets = facets;

  ccaSearch.initialView.tilesLandingPageParams.sections = [
    {
      id: 'types',
      title: 'Types',
      facetField: 'objectProvides',
      sortOn: 'alpha',
      icon: {
        family: 'Content types',
      },
    },
    {
      id: 'cca_geographic_countries',
      title: 'Countries',
      facetField: 'cca_geographic_countries.keyword',
      // filterType: 'any:exact',
      sortOn: 'alpha',
      // maxPerSection: 100,
      whitelist: [
        'Austria',
        'Belgium',
        'Bulgaria',
        'Croatia',
        'Cyprus',
        'Czechia',
        'Denmark',
        'Estonia',
        'Finland',
        'France',
        'Germany',
        'Greece',
        'Hungary',
        'Iceland',
        'Ireland',
        'Italy',
        'Latvia',
        'Liechtenstein',
        'Lithuania',
        'Luxembourg',
        'Malta',
        'Netherlands',
        'Norway',
        'Poland',
        'Portugal',
        'Romania',
        'Slovakia',
        'Slovenia',
        'Spain',
        'Sweden',
        'Switzerland',
        'TÃ¼rkiye',
        'Albania',
        'Bosnia and Herzegovina',
        'Kosovo',
        'Montenegro',
        'North Macedonia',
        'Serbia',
      ],
      icon: {
        family: 'CountryFlags',
        className: 'facet-option-icon',
      },
    },
  ];

  if (typeof window !== 'undefined') {
    config.searchui.ccaSearch.host =
      process.env.RAZZLE_ES_PROXY_ADDR || getClientProxyAddress();
  }
  let fieldsMatchAll = [
    'cca_climate_impacts.keyword',
    'cca_adaptation_sectors.keyword',
    'cca_adaptation_elements.keyword',
    'cca_key_type_measure.keyword',
  ];
  config.searchui.ccaSearch.facets.forEach((facet) => {
    if (fieldsMatchAll.includes(facet.field)) {
      facet.filterType = 'all';
    }
  });

  return config;
}
