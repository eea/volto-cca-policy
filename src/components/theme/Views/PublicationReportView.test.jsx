import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-intl-redux';
import PublicationReportView from './PublicationReportView';
import renderer from 'react-test-renderer';
import config from '@plone/volto/registry';

config.blocks = {
  blocksConfig: {
    title: {
      view: () => <div>Title Block Component</div>,
    },
  },
};

const mockStore = configureStore();

jest.mock('semantic-ui-react', () => ({
  ...jest.requireActual('semantic-ui-react'),
}));

describe('PublicationReportView', () => {
  it('should render the component', () => {
    const content = {
      title: 'My PublicationReportView',
      geochars:
        '{\r\n "geoElements":{"element":"GLOBAL",\r\n "macrotrans":null,"biotrans":null,"countries":[],\r\n "subnational":[],"city":""}}',
    };

    const store = mockStore({
      userSession: { token: '1234' },
      intl: {
        locale: 'en',
        messages: {},
      },
    });

    const component = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <PublicationReportView content={content} />
        </MemoryRouter>
      </Provider>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
