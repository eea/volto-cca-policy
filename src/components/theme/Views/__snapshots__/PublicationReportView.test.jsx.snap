// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PublicationReportView should render the component 1`] = `
<div
  className="db-item-view publication-report-view"
>
  <div>
    Title Block Component
  </div>
  <div
    className="ui container"
  >
    <h2>
      Description
    </h2>
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": undefined,
        }
      }
    />
    <div
      className="ui divider"
    />
    <h2>
      Reference information
    </h2>
    <div
      className="published-modified-info"
    >
      <p>
        <span>
          <strong>
            Published in Climate-ADAPT
          </strong>
           
          Invalid Date
        </span>
        <span>
             -   
        </span>
        <span>
          <strong>
            Last Modified in Climate-ADAPT
          </strong>
           
          Invalid Date
        </span>
      </p>
    </div>
    <div
      className="share-info"
    >
      <a
        href="/help/share-your-info"
      >
        <button
          className="ui button primary icon labeled"
        >
          Share your information
          <i
            aria-hidden="true"
            className="arrow right icon"
          />
        </button>
      </a>
    </div>
    <div
      className="content-box"
    >
      <div
        className="content-box-inner"
      >
        <div
          className="ui segment"
        >
          <div
            className="content-metadata"
          >
            <h5>
              Geographic characterisation:
            </h5>
            <div
              className="geochar"
            >
              <p>
                Global
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
